src/internal/group_api/mod.rs

```diff
@@ -359,88 +372,153 @@ pub(crate) fn get_group_keys<'a>(
 /// `user_master_pub_key` - public key of the user creating this group.
 /// `group_id` - unique id for the group within the segment.
 /// `name` - name for the group. Does not need to be unique.
-/// `members` - list of user ids to add as members of the group. This list takes priority over `add_as_member`,
-///     so the calling user will be added as a member if their id is in this list even if `add_as_member` is false.
+/// `members` - list of user ids to add as members of the group.
 /// `needs_rotation` - true if the group private key should be rotated by an admin, else false
 pub fn group_create<'a, CR: rand::CryptoRng + rand::RngCore>(
     recrypt: &'a Recrypt<Sha256, Ed25519, RandomBytes<CR>>,
     auth: &'a RequestAuth,
-    user_master_pub_key: &'a PublicKey,
     group_id: Option<GroupId>,
     name: Option<GroupName>,
-    members: &'a Vec<UserId>,
+    owner: Option<UserId>,
+    admins: Vec<UserId>,
+    members: Vec<UserId>,
+    users_to_lookup: &'a Vec<UserId>,
```

Comment:
I think this should be computed dynamically in this function, or if you add `admin_and_members` to `GroupCreateOpsStd`, perhaps you should just pass the `GroupCreateOpsStd` struct into the internal api.

Reply:
I think this will get easier with the switch to `async fn` as impl Future seems more temperamental...

Reply:
I'm hoping so